pkg/services/baremetal/host/host.go

```diff
@@ -1181,13 +1189,43 @@ func verifyConnectionRefused(sshClient sshclient.Client, port int) bool {
 	return true
 }
 
-func (s *Service) actionEnsureProvisioned() actionResult {
+func (s *Service) actionEnsureProvisioned() (ar actionResult) {
 	sshClient := s.scope.SSHClientFactory.NewClient(sshclient.Input{
 		PrivateKey: sshclient.CredentialsFromSecret(s.scope.OSSSHSecret, s.scope.HetznerBareMetalHost.Spec.Status.SSHSpec.SecretRef).PrivateKey,
 		Port:       s.scope.HetznerBareMetalHost.Spec.Status.SSHSpec.PortAfterCloudInit,
 		IP:         s.scope.HetznerBareMetalHost.Spec.Status.GetIPAddress(),
 	})
 
+	defer func() {
+		// Create an Event which contains the content of /var/log/cloud-init-output.log
+
+		if _, ok := ar.(actionContinue); ok {
```

Comment:
I think you should also ignore actionError. In that case we return an error message because of an unexpected error... 
Note that this does not include the case where we realize that cloud init failed, then we return actionFailed instead.

Reply:
I have concerns: Is it possible that the controller hangs in returning actionError until a MHC deletes the machine, and then we don't get the content of the file?